numpy/random/bit_generator.pyx

```diff
@@ -0,0 +1,499 @@
+    If a sequence of ints or strings, interpret each element as above and
+    concatenate.
+    Note that the handling of `int64` or `uint64` arrays are not just
+    straightforward views as `uint32` arrays. If an element is small enough to
+    fit into a `uint32`, then it will only take up one `uint32` element in the
+    output. This is to make sure that the interpretation of a sequence of
+    integers is the same regardless of numpy's default integer type, which
+    differs on different platforms.
+    Parameters
+    ----------
+    x : int, str, sequence of int or str
+    Returns
+    -------
+    seed_array : uint32 array
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from seed_seq import _coerce_to_uint32_array
+    >>> _coerce_to_uint32_array(12345)
+    array([12345], dtype=uint32)
+    >>> _coerce_to_uint32_array('12345')
+    array([12345], dtype=uint32)
+    >>> _coerce_to_uint32_array('0x12345')
+    array([74565], dtype=uint32)
+    >>> _coerce_to_uint32_array([12345, '67890'])
+    array([12345, 67890], dtype=uint32)
+    >>> _coerce_to_uint32_array(np.array([12345, 67890], dtype=np.uint32))
+    array([12345, 67890], dtype=uint32)
+    >>> _coerce_to_uint32_array(np.array([12345, 67890], dtype=np.int64))
+    array([12345, 67890], dtype=uint32)
+    >>> _coerce_to_uint32_array([12345, 0x10deadbeef, 67890, 0xdeadbeef])
+    array([     12345, 3735928559,         16,      67890, 3735928559],
+          dtype=uint32)
+    >>> _coerce_to_uint32_array(1234567890123456789012345678901234567890)
+    array([3460238034, 2898026390, 3235640248, 2697535605,          3],
+          dtype=uint32)
+    """
+    if isinstance(x, np.ndarray) and x.dtype == np.dtype(np.uint32):
+        return x.copy()
+    elif isinstance(x, str):
+        if x.startswith('0x'):
+            x = int(x, base=16)
+        elif DECIMAL_RE.match(x):
+            x = int(x)
+        else:
+            raise ValueError("unrecognized seed string")
+    if isinstance(x, (int, np.integer)):
+        return _int_to_uint32_array(x)
+    elif isinstance(x, (float, np.inexact)):
+        raise TypeError('seed must be integer')
+    else:
+        if len(x) == 0:
+            return np.array([], dtype=np.uint32)
+        # Should be a sequence of interpretable-as-ints. Convert each one to
+        # a uint32 array and concatenate.
+        subseqs = [_coerce_to_uint32_array(v) for v in x]
+        return np.concatenate(subseqs)
+
+
+cdef uint32_t hashmix(uint32_t value, uint32_t * hash_const):
+    # We are modifying the multiplier as we go along, so it is input-output
+    value ^= hash_const[0]
+    hash_const[0] *= MULT_A
+    value *=  hash_const[0]
+    value ^= value >> XSHIFT
+    return value
+
+cdef uint32_t mix(uint32_t x, uint32_t y):
+    cdef uint32_t result = (MIX_MULT_L * x - MIX_MULT_R * y)
+    result ^= result >> XSHIFT
+    return result
+
+
+cdef class SeedSequence():
```

Comment:
Does SeedSequence need any customized support for pickling?

Reply:
Well, it's a requirement that it be pickleable, of course. I don't know if Cython is providing that out-of-box correctly. Should be tested, of course.

Reply:
Nope, it just works:
```
>>> ss = np.random.SeedSequence(entropy=range(10), 
...        program_entropy=range(20), spawn_key=(2,), pool_size=10)
>>> aa = pickle.loads(pickle.dumps(ss))
>>> aa.state
{'entropy': range(0, 10), 'program_entropy': range(0, 20),
   'spawn_key': (2,), 'pool_size': 10}
>>> aa.state == ss.state
True
```

Reply:
test added